.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,209 @@
+```  
+
+Run the component tests:
+```sh
+npm run test:component
+```
+
+You must write corresponding unit and component tests for any code you add or modify, and all tests must pass before those changes can be merged back to the master branch.
+
+#### Functional tests
+This repo also contains the [functional tests][functional-tests] for the Azure DevOps ShellCheck Extension. These are primarily only used from the release pipeline that deploys the extension to the [Marketplace][vs-marketplace]. 
+
+These tests trigger Azure Pipelines builds that utilize the ShellCheck tasks to validate the published tasks are functioning correctly. Those pipelines are defined in our [ShellCheck Tests Repo][shellcheck-tests-repo] which also contains the sample shell scripts that are used for the ShellCheck scans.
+
+The functional tests can be invoked via:
+```sh
+npm run test:functional
+```
+
+However, you most likely will never need to run these functional tests from your local workspace. 
+
+### Code coverage
+Code coverage is generated, and enforced using [Istanbuljs/nyc][nyc-url]. The unit test suite has 100% coverage of the application source code, and similarly the component test suite also has 100% coverage. We will not accept any changes that drop the code coverage below 100%.
+
+The tests will be run as part of the npm `build` script and on a git commit, but there are also several coverage related `npm scripts` that you can run to generate and/or view the coverage details.  
+
+Generate latest coverage for the unit tests:
+```sh
+npm run coverage
+```  
+
+Open the detailed HTML coverage report for the unit tests:
+```sh
+npm run coverage:open
+```
+
+Generate latest coverage for the component tests:
+```sh
+npm run coverage:component
+```  
+
+Open the detailed HTML coverage report for the component tests:
+```sh
+npm run coverage:component:open
+```
+
+### Linting
+This repo uses [tslint][tslint-url] and [eslint][eslint-url] for respectively linting the TypeScript and JavaScript code. The tslint configuration can be found in the [tslint.json][tslint-config-url] file and the eslint configuration can be found in the [.eslintrc.js][eslint-config-url] file.
+
+Both tslint and eslint are automatically run when you run the npm `build` script and when you make a commit. Additionally you can run also run the linters via their corresponding npm script.
+
+For example to run both eslint and tslint:
+```sh
+npm run lint
+```  
+
+Or to just run [tslint][tslint-url]:
+```sh
+npm run tslint
+```  
+
+Or to just run [eslint][eslint-url]:
+```sh
+npm run eslint
+```  
+
+### Versioning
+All of the manifest files associated with this extension (the task manifests, the extension manifest, and the package manifest) have their versions automatically incremented (patch version is bumped) by the relevant scripts and/or release process. As such there is no need to manually update the versions in most circumstances. 
+
+When you should manually update a manifest version:
```

Comment:
I might look into making this automated with https://github.com/semantic-release/semantic-release